src/cyclebane/graph.py

```diff
@@ -51,72 +43,69 @@ def _remove_ancestors(graph: nx.DiGraph, node: Hashable) -> nx.DiGraph:
 
 @dataclass(frozen=True)
 class IndexValues:
-    axes: tuple[IndexName]
-    values: tuple[IndexValue]
+    """
+    Index values used as part of :py:class:`NodeName`.
+
+    Conceptually, this is a mapping from index names to index values.
+    """
+
+    axes: tuple[IndexName, ...]
+    values: tuple[IndexValue, ...]
 
     @staticmethod
-    def from_tuple(t: tuple[tuple[IndexName, IndexValue]]) -> IndexValues:
+    def from_tuple(t: tuple[tuple[IndexName, IndexValue], ...]) -> IndexValues:
         names = tuple(name for name, _ in t)
         values = tuple(value for _, value in t)
         return IndexValues(axes=names, values=values)
 
-    def to_tuple(self) -> tuple[tuple[IndexName, IndexValue]]:
+    def to_tuple(self) -> tuple[tuple[IndexName, IndexValue], ...]:
         return tuple(zip(self.axes, self.values))
 
     def merge_index(self, other: IndexValues) -> IndexValues:
         return IndexValues(
             axes=other.axes + self.axes, values=other.values + self.values
         )
 
-    def pop(self, name: IndexName) -> IndexValues:
-        i = self.axes.index(name)
-        return IndexValues(
-            axes=self.axes[:i] + self.axes[i + 1 :],
-            values=self.values[:i] + self.values[i + 1 :],
-        )
-
-    def pop_axis(self, axis: int) -> IndexValues:
-        if axis < 0 or axis >= len(self.axes):
-            raise ValueError('Invalid axis')
-        return IndexValues(
-            axes=self.axes[:axis] + self.axes[axis + 1 :],
-            values=self.values[:axis] + self.values[axis + 1 :],
-        )
-
-    def __str__(self):
+    def __str__(self) -> str:
         return ', '.join(
             f'{name}={value}' for name, value in zip(self.axes, self.values)
         )
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self.axes)
 
 
 @dataclass(frozen=True)
```

Comment:
These classes are used a bunch so they may benefit from
```suggestion
@dataclass(frozen=True, slots=True)
```